[WIP] Adds cluster_name parameter to listRecommendations API

shreyabiradar07 opened the pull request:
This PR fixes #940 

-  Enhancement of `/listRecommendations` API by adding `cluster_name` query parameter to retrieve recommendations for a specific cluster based on its name

khansaad commented on src/main/java/com/autotune/database/service/ExperimentDBService.java:
```diff
@@ -189,6 +189,53 @@ public void loadRecommendationsFromDBByName(Map<String, KruizeObject> mainKruize
         }
     }
+    // Todo - confirm if experimentResultData to be changed to clusterResultData
+    public void loadResultsFromDBByClusterName(Map<String, KruizeObject> mainKruizeExperimentMap, String clusterName, Timestamp interval_end_time, Integer limitRows) throws Exception {
```
This method isn't required as well, can be removed.

khansaad commented on src/main/java/com/autotune/database/service/ExperimentDBService.java:
```diff
@@ -297,6 +344,34 @@ public void loadExperimentFromDBByName(Map<String, KruizeObject> mainKruizeExper
+                    }
+                }
+                if (failureThreshHold > 0 && failureCount == failureThreshHold) {
+                    throw new Exception("Experiments of cluster " + clusterName + " unable to load from DB.");
```
Change this exception message so that it makes more sense

khansaad commented on src/main/java/com/autotune/database/service/ExperimentDBService.java:
```diff
@@ -313,6 +388,16 @@ public void loadExperimentAndRecommendationsFromDBByName(Map<String, KruizeObjec
         loadRecommendationsFromDBByName(mainKruizeExperimentMap, experimentName);
     }
 
+    public void loadExperimentsAndResultsFromDBByClusterName(Map<String, KruizeObject> mainKruizeExperimentMap, String clusterName) throws Exception {
```
Remove this as well.

shreyabiradar07 commented on src/main/java/com/autotune/analyzer/services/ListRecommendations.java:
```diff
@@ -145,7 +146,60 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response) t
                     );
                 }
-            } else {
+            }else if(null != clusterName){
```
Done

shreyabiradar07 commented:
@khansaad - I have incorporated review comments and rebased branch 'add_clusterName_parameter' onto 'mvp_demo'. 
 Can you please review the updated PR?

dinogun commented:
Closing stale PRs
